ff/serving/metadata/metadata.go

```diff
@@ -140,7 +140,7 @@ type Resource interface {
 	ID() ResourceID
 	Dependencies(ResourceLookup) (ResourceLookup, error)
 	Proto() proto.Message
-	UpdateStatus(ResourceStatus) error
```

Comment:
can we not make this a string, and could we make the types also exist in the proto files. Let's avoid using strings when we mean Enums.